loader/cJSON.c

```diff
@@ -1215,3 +1222,133 @@ void cJSON_Minify(char *json) {
     }
     *into = 0; /* and null-terminate. */
 }
+
+VkResult loader_get_json(const struct loader_instance *inst, const char *filename, cJSON **json) {
+    FILE *file = NULL;
+    char *json_buf = NULL;
+    size_t len;
+    VkResult res = VK_SUCCESS;
+
+    assert(json != NULL);
+
+    *json = NULL;
+
+#if defined(_WIN32)
+    int filename_utf16_size = MultiByteToWideChar(CP_UTF8, 0, filename, -1, NULL, 0);
+    if (filename_utf16_size > 0) {
+        wchar_t *filename_utf16 = (wchar_t *)loader_stack_alloc(filename_utf16_size * sizeof(wchar_t));
+        if (MultiByteToWideChar(CP_UTF8, 0, filename, -1, filename_utf16, filename_utf16_size) == filename_utf16_size) {
+            file = _wfopen(filename_utf16, L"rb");
+        }
+    }
+#elif COMMON_UNIX_PLATFORMS
+    file = fopen(filename, "rb");
+#else
+#warning fopen not available on this platform
```

Comment:
Why isn't this an `#error`?

Reply:
Consistency with the rest of the codebase. #error is used for 'maintainers of this repo have messed up', while #warning is for places new platforms need to add code for their platform.